### ISSUE TITLE
Add an automated accessibility test to the CI pipeline 

### ISSUE THREAD

**Issue by bk3c** [bloom-housing/bloom#142]
The Storybook A11y plugin uses https://github.com/dequelabs/axe-core to provide accessibility feedback from within SB, and it looks like we could also extend it pretty easily to test as part of the regular pipeline.

- [ ] Get the basic testing working and a couple components tested as part of each CI run
- [ ] Test all of ui-components and fix any issues so we can move forward clean by default
- [ ] Test all of the public app and fix any issues there too

I'm definitely open to other a11y testing solutions if there's a better / more comprehensive option out there, but this is the first very promising one I've found that looks ready for CI.

**Comment by bk3c** [comment]
@akegan I know this is a current topic in your neck of the woods, so if you have any feedback one way or the other as to whether this is a good solution from the accessibility experts over there I'd be glad to hear it.

**Comment by akegan** [comment]
It looks like a good start at least. At first glance I don't see anything [in their list of what they cover](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md) about logical reading order (headers that make sense with a screen reader, only one h1 tag per page, no jumps from e.g. h1 to h3) or size of text, but definitely seems like a good start. 

We haven't settled on an accessibility linter yet, but our goal would be to be able to use a consistent one across multiple projects if possible, and to have it included in CI.

**Comment by bk3c** [comment]
In addition to running axe in CI, we should also look at adding https://github.com/jsx-eslint/eslint-plugin-jsx-a11y to our eslint rule set. It looks like it provides a subset of checks, but better to have at least some feedback sooner?

**Comment by akegan** [comment]
@seanmalbert are you up for me picking up this ticket and trying to add a11y testing for UI components (making circleCI fail when there are violations? [This blog post](https://blog.atrera.com/javascript/pipeline/2021/03/10/a-guide-to-add-auto-accessibility-check-with-storybook.html) makes it look like it'd be pretty simple. We can see how many errors we get and try to fix some, or not make the tests blocking to start. 

[Sample component](https://storybook.bloom.exygy.dev/?path=/story/actions-button--inaccessible) that has an a11y violation now:
![image](https://user-images.githubusercontent.com/11825994/124016935-fd223300-d99a-11eb-8219-1c46841b0aba.png)

**Comment by akegan** [comment]
@seanmalbert @jesse I've got automated a11y testing for UI-components up and running locally, and am planning to work on incorporating it into CircleCI next. I used the model from the blog post linked above, which involves re-adding the storyshots library, but does not require actually generating the storyshot snapshots. 

The tests just basically raise an error for every story that has a violation, with the violations being the same as the ones you see in the UI. It's also possible in cases where the story or component is intentionally inaccessible (like in the inaccessible button case above) or the color contrast is slightly under 4.5:1 we can disable specific rules for specific components. 

Running the tests now, there are 31 failing stories, which are a mix of mostly color contrast issues, a couple of "Ensure image alternative is not repeated as text" and a couple of missing alt texts. I can fix the missing alt text and repetitive alt text, but I think I need to work with @slowbot to go through the color contrast issues. The output of the tests isn't super legible, but I'll work on getting a neater list together of the components that axe is flagging for contrast issues.

**Comment by slowbot** [comment]
@akegan Let me know how I can assist here. Happy to pull your branch locally and work on contrast issues

**Comment by akegan** [comment]
Thanks @slowbot, if you pull down the branch from this PR, and change color-contrast back to `enabled: true` on [this line](https://github.com/bloom-housing/bloom/pull/1450/files#diff-7be74b34cc1d6475298bf54814896cf7e4a0b0860e5244851ffd5d0b6bbef6edR33). you should be able to run `yarn test:a11y` from the UI-components repo and see the issues. Let me know if you run into any issues or if anything is confusing, I'm happy to pair on it!

If it makes more sense, we can also keep color-contrast disabled for now and then follow-up with a separate PR to fix color contrast issues.
